File: Programmers/야근 지수.java
import java.util.Comparator;
import java.util.PriorityQueue;

class Solution {
    public long solution(int n, int[] works) {
        long answer = 0;
        PriorityQueue<Integer> pq = new PriorityQueue<>(Comparator.reverseOrder());
        for (int i : works) pq.add(i);
        while (n-- > 0) {
            if (pq.peek() > 0) {
                pq.add(pq.poll() - 1);
            }
        }
        while (!pq.isEmpty()) {
            int num = pq.poll();
            answer += (num * num);
        }
        return answer;
    }
}

File: Programmers/야근 지수.cpp
#include <string>
#include <vector>
#include <queue>
using namespace std;

long long solution(int n, vector<int> works) {
    long long answer = 0;
    priority_queue<int> pq;
    for (int i : works) pq.push(i);
    while (n--) {
        int tmp = pq.top();
        if (tmp > 0) {
            pq.pop();
            pq.push(tmp - 1);
        }
    }
    while (!pq.empty()) {
        int tmp = pq.top();
        pq.pop();
        answer += (tmp * tmp);
    }
    return answer;
}